website/examples/date-picker-dialog.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { DayPicker } from 'react-day-picker';
+import { usePopper } from 'react-popper';
+
+import { format, isValid, parse } from 'date-fns';
+import FocusTrap from 'focus-trap-react';
+
+export default function Example() {
+  const [selected, setSelected] = React.useState<Date>();
+  const [inputValue, setInputValue] = React.useState<string>('');
+  const [isOpen, isOpenSet] = React.useState(false);
+
+  const refEl = React.useRef<HTMLDivElement>(null);
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+  const [popperEl, popperElSet] =
+    React.useState<HTMLDivElement | null>(null);
+
+  const closePopper = () => {
+    isOpenSet(false);
+    buttonRef?.current?.focus();
+  };
+
+  const { styles, attributes } = usePopper(
+    refEl.current,
+    popperEl,
+    {
+      placement: 'bottom-start'
+    }
+  );
+
+  const dateFormat = 'P';
+  const formatDate = (date: Date) => format(date, dateFormat);
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const text = event.currentTarget.value;
+    setInputValue(text);
+    const date = parse(text, dateFormat, new Date());
+    if (isValid(date)) {
+      setSelected(date);
+    } else {
+      setSelected(undefined);
+    }
+  };
+
+  const handleDaySelected = (date: Date | undefined) => {
+    setSelected(date);
+    if (date) {
+      setInputValue(formatDate(date));
+      closePopper();
+    } else {
+      setInputValue('');
+    }
+  };
+
+  return (
+    <div style={{ position: 'relative' }}>
+      <div
+        ref={refEl}
+        style={{
+          padding: '4px 0',
+          display: 'flex',
+          alignItems: 'center'
+        }}
+      >
+        Date
+        <input
+          type="text"
+          style={{
+            marginLeft: '4px',
+            padding: '6px 4px',
+            lineHeight: '24px',
+            fontSize: '16px',
+            height: '40px'
```

Comment:
🤔 Thinking how to remove these style clutter and reduce the example number of lines..

Reply:
It's hard to preview the website's css...I think there is some, but it doesn't load when I escape the sandpack context...